app/controllers/statistics_controller.rb

```diff
@@ -12,6 +14,31 @@ def collection
     @all_indexers     = build_user_array(DeedType::PAGE_INDEXED)
   end
 
+  def export_csv
+    header = ['User Login', 'User Name', 'Email', 'Opt-In']
+    owner = User.find 'ushmmarchives'
+    collection_ids = owner.collections.map { |c| c.id }.sort
+    deed_map = Deed.where(:collection_id => collection_ids).group(:user_id, :collection_id).count
+    user_ids = deed_map.keys.map {|e| e[0]}.uniq
+    Collection.where(:id => collection_ids).order(:id).each { |c| header << c.title }
+
+    rows = []
+
+    User.find(user_ids).each do |user|
+      rows << [user.login, user.display_name, user.email, user.activity_email]
+    end
```

Comment:
Add all the users to the rows array, but only with the first four columns.

(How are we going to figure out which user belongs to which set of deeds?  There is no `order` clause, so I worry that we're going to mix things up.)

Reply:
Okay, the original sample code in #1588 includes this:
```

  collection_ids.each do |collection_id|
    row << deed_map[[user.id,collection_id]] || 0
  end
```
This loops through each of the collections for the user (preserving the order we used earlier)
Then it constructs a key for the `deed_map` hash out of the ID for this row's user and the ID for this column's collection.  
This `[user.id, collection.id]` key should return a deed count from `deed_map`, which is appended as a new cell onto this user's row.

Reply:
@benwbrum this was an oversight on my part, thanks for your help :+1: